wsmaster/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java

```diff
@@ -185,6 +188,14 @@ protected static class FactoryParametersResolverHolder {
      */
     public FactoryParametersResolver getFactoryParametersResolver(Map<String, String> parameters)
         throws BadRequestException {
+      String url = parameters.get(URL_PARAMETER_NAME);
+      // Check if the URL is a raw devfile URL. If so, use the default resolver,
+      // which resolves factories from a direct URL to a devfile content.
+      if (!isNullOrEmpty(url)
+          && devfileFilenamesProvider.getConfiguredDevfileFilenames().stream()
```

Comment:
Why `defaultFactoryResolver.accept(parameters)` doesn't work here?

Reply:
Moving the `devfileFilenamesProvider.getConfiguredDevfileFilenames().stream().anyMatch(url::endsWith)` check to the `defaultFactoryResolver.accept(parameters)` check.

Reply:
@tolusha Moved the logic to the `defaultFactoryResolver.accept(parameters)` please take a look.